src/App.tsx

```diff
@@ -1,26 +1,55 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { FC, useEffect, useState } from 'react';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { getTodos } from './api';
+import { getFilteredData } from './utils/getFilteredData';
+import { useAppDispatch, useAppSelector } from './app/hook';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App: FC = () => {
+  const [loading, setLoading] = useState(false);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const dispatch = useAppDispatch();
+  const openedTodo = useAppSelector(state => state.currentTodo);
+  const todos = useAppSelector(state => state.todos);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(res => dispatch(todosSlice.actions.setTodos(res)))
+      .finally(() => setLoading(false));
+  }, [dispatch]);
```

Comment:
Consider adding error handling for the `getTodos` promise to manage potential API errors gracefully. This will improve the user experience by providing feedback in case of a failure.